x-pack/test/api_integration/apis/endpoint/resolver.ts

```diff
@@ -222,10 +219,28 @@ export default function resolverAPIIntegrationTests({ getService }: FtrProviderC
           .get(`/api/endpoint/resolver/${entityID}/children?legacyEndpointID=foo`)
           .set(commonHeaders)
           .expect(200);
-        expect(body.pagination.total).to.eql(0);
-        expect(body.pagination.next).to.eql(null);
+        expect(body.pagination.nextChild).to.eql(null);
         expect(body.children).to.be.empty();
       });
     });
+
+    describe('tree endpoint', () => {
```

Comment:
I can probably implement it in a follow up PR but we should do more testing. Currently we only test using the legacy stuff and not `entity_id`. We should also write some unit tests for the recursive functionality and leverage the resolver generator to build the tree out for us.